metabase/metabase
# Fix - App crashes when navigating back during model creation

Closes #34514


### Description

There were 2 issues:
1. Navigating between "Query" and "Metadata" tabs in model editor used `history.replace` instead of `history.push`, even though they have different pathnames (`/model/query` & `/model/metadata`)
2. When navigating back to the step before the data source is selected there was an invalid POST `/api/dataset` request being sent (empty query: `{ "query": {}, "type": "query", "parameters": [] }`)

### How to verify

Stress test: https://github.com/metabase/metabase/actions/runs/9974686269

----

1. New > Model > Use the notebook editor > Orders
2. Run the question
3. Go to "Metadata" tab
4. Click the back button in your browser

"Query" tab should be active now

5. Click the back button in your browser
6. Close the entity picker

There should be no error message.
The state should look exactly the same as when you go to New > Model > Use the notebook editor, and close the entity picker modal.
